### PR TITLE
Upgrade to new spago

### PR DESCRIPTION
**Pull request by CGenie**
This PR contains an upgrade of this package to the new spago with `spago.yaml` replacing `spago.dhall`, `packages.dhall`, `bower.json`.

### CODE REVIEW

File: yarn.lock
Comment (fsoikin) [inline yarn.lock:1]: Why is this file necessary?

File: shell.nix
Comment (fsoikin) [inline shell.nix:1]: Why is this file necessary?

File: spago.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+workspace:
+  packageSet:
+    registry: 50.7.0
+  extraPackages: {}
+
+package:
+  name: spec-discovery
+  dependencies:
```
Comment (fsoikin) [inline spago.yaml:8]: Do the dependencies have to be explicitly versioned? I see the weren't versioned in the old spago config.

File: spago.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+workspace:
+  packageSet:
+    registry: 50.7.0
+  extraPackages: {}
```
Comment (fsoikin) [inline spago.yaml:4]: This can probably be omitted, right?

File: spago.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  dependencies:
+    - aff: ">=7.1.0 <8.0.0"
+    - aff-promise: ">=4.0.0 <5.0.0"
+    - effect: ">=4.0.0 <5.0.0"
+    - foldable-traversable: ">=6.0.0 <7.0.0"
+    - prelude: ">=6.0.1 <7.0.0"
+    - spec: ">=7.6.0 <8.0.0"
+  test:
+    main: Test.Main
+    dependencies: []
```
Comment (fsoikin) [inline spago.yaml:17]: And this I guess?

File: spago.yaml (hunk 1 above)
Comment (CGenie) [reply spago.yaml:8]: No, they weren't, but I think it's a good practice to specify version numbers nevertheless.

In Haskell you have things like: https://hackage.haskell.org/package/cabal-constraints which pin the versions exactly.

If spec releases version 8.x.x, someone will soon discover that `purescript-spec-discovery` won't compile because of dependency errors and would try to fix it to work with newer version.

Comment (fsoikin) [reply spago.yaml:8]: Isn't the whole point of package sets to avoid these things? 

For example, if there is a new version of spec that makes discovery not build anymore, then discovery won't be included in the next package set. And anybody who needs it can just stay with the previous package set version, not upgrade to the new one.

Comment (CGenie) [reply spago.yaml:8]: Yeah, maybe you're right, I'll fix these.

Comment (CGenie) [reply spago.yaml:8]: Ah no sorry, I forgot that `spago publish` requires these things to be added. When I didn't specify the versions, it complained that it can't publish such a package.

Comment (fsoikin) [reply spago.yaml:8]: Ok, good for now. I'll look into this some more later.

**Comment by fsoikin** [comment]
@CGenie I think you need to update the GHA config too. See how GHA is failing?

**Comment by CGenie** [comment]
```
import fs15, { constants as fsConstants } from "node:fs/promises";
               ^^^^^^^^^
SyntaxError: The requested module 'node:fs/promises' does not provide an export named 'constants'
```
I guess we need nodejs at least 16, it's 14 in tests still. https://en.wikipedia.org/wiki/Node.js#Releases I guess version 22 is a safe bet now?

**Comment by fsoikin** [comment]
Still failing

**Comment by CGenie** [comment]
Yeah because I forgot about bower.

**Comment by CGenie** [comment]
I don't understand github. I wanted to set up my own runner for this but I don't know how.

**Comment by CGenie** [comment]
Thanks!
